packages/go_router/lib/src/redirection.dart

```diff
@@ -71,13 +71,21 @@ RouteMatchList redirect(RouteMatchList prevMatchList,
     if (top == null) {
       break;
     }
-    final String? topRouteLocation = top.route.redirect(
+
+    final RouteBase topRoute = top.route;
+    if (topRoute is! GoRoute) {
+      throw RedirectionException(
+          'Last RouteMatch should contain a GoRoute, but was ${topRoute.runtimeType}',
+          <RouteMatchList>[matches],
+          uri);
+    }
+    final String? topRouteLocation = topRoute.redirect(
       GoRouterState(
         configuration,
         location: currentMatches.location.toString(),
         subloc: top.subloc,
-        name: top.route.name,
-        path: top.route.path,
+        name: (top.route as GoRoute).name,
```

Comment:
Would be better to cast it to a goroute variable first

Reply:
1288344455debd39d2723fd57ec4dbce75b12f7f